providers/default/defltprov.c

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2019 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <string.h>
+#include <stdio.h>
+#include <openssl/core.h>
+#include <openssl/core_numbers.h>
+#include <openssl/core_names.h>
+#include <openssl/params.h>
+
+/* Functions provided by the core */
+static OSSL_core_get_param_types_fn *c_get_param_types = NULL;
+static OSSL_core_get_params_fn *c_get_params = NULL;
+
+/* Parameters we provide to the core */
+static const OSSL_ITEM deflt_param_types[] = {
+    { OSSL_PARAM_UTF8_PTR, OSSL_PROV_PARAM_NAME },
+    { OSSL_PARAM_UTF8_PTR, OSSL_PROV_PARAM_VERSION },
+    { OSSL_PARAM_UTF8_PTR, OSSL_PROV_PARAM_BUILDINFO },
+    { 0, NULL }
+};
+
+static const OSSL_ITEM *deflt_get_param_types(const OSSL_PROVIDER *prov)
+{
+    return deflt_param_types;
+}
+
+static int deflt_get_params(const OSSL_PROVIDER *prov,
+                            const OSSL_PARAM params[])
+{
+    return OSSL_PARAM_optional_set_utf8_ptr(params,
+                                            OSSL_PROV_PARAM_NAME,
+                                            "OpenSSL Default Provider")
+           && OSSL_PARAM_optional_set_utf8_ptr(params,
```

Comment:
This stops if one fails.  Don't you want to try to do them all?

Reply:
I'd say no. These function succeed if a given parameter is absent from from params, so a failure would have to be something like the datatype doesn't match what we're expecting. That's probably a legitimate failure because it shouldn't happen.

Reply:
Agreed.  If one of the OSSL_PARAM items has the wrong data_type, the get_params operation should fail.  Sure, it means that *some* of them still have values, but that's inconsequential.

Reply:
Resolving with the closing of #8519